### PR TITLE
Add unit and integration tests for monitoring module

### PR DESCRIPTION
**Pull request by gimballock**
Add test coverage for the monitoring/metrics system at both the unit and integration level.

**Changes**

**Commit 1**: Unit tests for monitoring data layer and HTTP API (1e45c730)

Unit tests for the previously untested monitoring module in stratum-apps/src/monitoring/:

- **client.rs**: Sv2ClientInfo accessors, to_metadata(), Sv2ClientsMonitoring default impls (get_sv2_client_by_id, get_sv2_clients_summary)
- **server.rs**: ServerInfo accessors with Option<f32> hashrate handling, ServerMonitoring default impl (get_server_summary)
- **sv1.rs**: Sv1ClientsMonitoring default impls (get_sv1_client_by_id, get_sv1_clients_summary including None-hashrate filtering)
- **prometheus_metrics.rs**: Conditional metric registration per role type, Prometheus text encoding round-trip
- **http_server.rs**: Pagination unit tests (effective_limit clamping, paginate edge cases), HTTP endpoint integration tests using tower::oneshot against mock monitoring sources for all 9 API routes and /metrics

Adds tower, http-body-util, hyper as dev-dependencies for the axum handler tests.

**Commit 2**: Integration test infrastructure and metrics assertions (1a6ef0ea)

- **Config setters** (pool/config.rs, translator/config.rs, jd-client/config.rs): Add set_monitoring_address() and set_monitoring_cache_refresh_secs() so tests can enable monitoring on dynamic ports with a fast 1-second cache refresh.

- **Test helpers** (integration-tests/lib/mod.rs): Add start_pool_with_monitoring(), start_jdc_with_monitoring(), start_sv2_translator_with_monitoring() wrappers that configure monitoring and return the monitoring address.

- **Assertion helpers** (integration-tests/lib/metrics_assert.rs): Async helpers (fetch_metrics, fetch_api, assert_api_health, assert_uptime, assert_metric_gte, assert_metric_eq, assert_metric_present, assert_metric_not_present) using spawn_blocking internally for the synchronous minreq HTTP calls. 11 unit tests for the parsing/assertion logic.

- **Metrics assertions** augment 9 existing integration test functions across 8 test files. Each test waits 2 seconds for the SnapshotCache to refresh, then validates topology-appropriate metrics:

| Test | Metrics validated |
|------|------------------|
| sv2_mining_device_and_pool_success | Pool: clients >= 1, uptime, health OK |
| pool_standard_channel_receives_share | Pool: clients >= 1, shares accepted >= 1, uptime, health OK |
| translate_sv1_to_sv2_successfully | Pool: clients >= 1, no server channels. tProxy: server channels present, SV1 clients >= 1, no SV2 clients, server shares accepted |
| test_extension_negotiation_with_tlv_in_submit_shares | Pool: clients >= 1, uptime, health OK |
| jd_non_aggregated_tproxy_integration | tProxy: server channels, SV1 clients >= 1. JDC: server channels, no SV2 clients. Pool: clients >= 1 |
| jd_aggregated_tproxy_integration | Pool: clients >= 1. tProxy: exactly 1 server channel (aggregated) |
| jdc_submit_shares_success | Pool: clients >= 1, uptime, health OK |
| propagated_from_jdc_to_tp | Pool: clients >= 1, shares accepted >= 1 |
| propagated_from_jds_to_tp | Pool: clients >= 1, shares accepted >= 1 |

**Technical notes**

- Metric helpers use `tokio::task::spawn_blocking` to wrap synchronous `minreq` HTTP calls, preventing deadlocks on the single-threaded tokio test runtime
- Tests use `#[tokio::test]` (current_thread) rather than `multi_thread`, because `FallbackHandler::drop()` in `fallback_coordinator.rs` panics if `done()` was never called — aborting translator/JDC tasks on a multi_thread worker triggers a double-panic SIGABRT
- Cache refresh is set to 1 second in tests (vs 15s default), keeping the per-test overhead to ~2 seconds

### CODE REVIEW

**Comment by gimballock** [comment]
I added some additional tests, let me know if you see any areas that still need more coverage

**Comment by GitGab19** [comment]
I would simply add the `enable_monitoring` parameter to the original starter methods (`start_jdc`, `start_translator`, etc.). So that we keep having a single method per app, and we don't need to call the `start_*_with_monitoring(..,true)`.

File: integration-tests/tests/monitoring_integration.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+            MessageDirection::ToUpstream,
+            MESSAGE_TYPE_OPEN_STANDARD_MINING_CHANNEL,
+        )
+        .await;
+    sniffer
+        .wait_for_message_type(
+            MessageDirection::ToDownstream,
+            MESSAGE_TYPE_OPEN_STANDARD_MINING_CHANNEL_SUCCESS,
+        )
+        .await;
```
Comment (GitGab19) [inline integration-tests/tests/monitoring_integration.rs]: Do we really need all these assertions?

We are only interested in asserting that a share is submitted, right?

File: stratum-apps/src/monitoring/client.rs [hunk 2]
```diff
@@ -135,3 +135,160 @@ pub trait Sv2ClientsMonitoring: Send + Sync {
...
+            42,
+            vec![make_extended(1, 100.0)],
+            vec![make_standard(2, 50.0), make_standard(3, 75.0)],
+        );
+        let meta = client.to_metadata();
+
+        assert_eq!(meta.client_id, 42);
+        assert_eq!(meta.extended_channels_count, 1);
+        assert_eq!(meta.standard_channels_count, 2);
+        assert!((meta.total_hashrate - 225.0).abs() < f32::EPSILON);
```
[the remark this answers is not in the document]
Comment (gimballock) [reply stratum-apps/src/monitoring/client.rs]: done

File: stratum-apps/src/monitoring/http_server.rs [hunk 3]
```diff
@@ -925,3 +925,621 @@ async fn handle_prometheus_metrics(State(state): State<ServerState>) -> Response
             .into_response(),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use axum::body::Body;
+    use http_body_util::BodyExt;
+    use tower::ServiceExt;
```
[the remark this answers is not in the document]
Comment (gimballock) [reply stratum-apps/src/monitoring/http_server.rs]: Yes — BodyExt brings `.collect()` (used in `get_body()`) and `ServiceExt` brings `.oneshot()` (used in every test to dispatch a request through the router). They're trait imports for method resolution.
